packages/block-library/src/calendar/edit.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * External dependencies
+ */
+import moment from 'moment';
+import memoize from 'memize';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	Disabled,
+	ServerSideRender,
+} from '@wordpress/components';
+import { Component } from '@wordpress/element';
+import { withSelect } from '@wordpress/data';
+
+class CalendarEdit extends Component {
+	constructor() {
+		super( ...arguments );
+		this.getYearMonth = memoize(
+			this.getYearMonth.bind( this ),
+			{ maxSize: 1 }
+		);
+		this.getServerSideAttributes = memoize(
+			this.getServerSideAttributes.bind( this ),
+			{ maxSize: 1 }
+		);
+	}
+
+	getYearMonth( date ) {
+		if ( ! date ) {
+			return {};
+		}
+		const momentDate = moment( date );
+		return {
+			year: momentDate.year(),
+			monthNum: momentDate.month() + 1,
+		};
+	}
+
+	getServerSideAttributes( attributes, date ) {
+		return {
+			...attributes,
+			...this.getYearMonth( date ),
```

Comment:
Why not set it on the server using WP functions?

Reply:
When the user changes the publish date, even if the post is not yet saved we want the user to see the calendar of that publish date so we need a way to pass that information to the server, that exactly what we are doing here.
The attributes month and year are never saved, they are just transient attributes used by the server side render mechanism.

Reply:
I still think it isn't a good approach. Let me explain on the post from my personal website:
https://gziolo.pl/2017/11/07/starter-kit-and-reusable-scripts/
If I visit it as of today you will see that the calendar in the sidebar renders the same month as when this post was published:
![screen shot 2019-02-14 at 11 29 11](https://user-images.githubusercontent.com/699132/52780997-c6f2eb80-304b-11e9-9f88-6298e9029ec3.png)

it behaves differently for the pages as it simply renders the current month:
![screen shot 2019-02-14 at 11 31 31](https://user-images.githubusercontent.com/699132/52781231-526c7c80-304c-11e9-8ec3-124735e7ee29.png)

There are also archive pages which follow the URL path:

![screen shot 2019-02-14 at 11 33 46](https://user-images.githubusercontent.com/699132/52781286-7d56d080-304c-11e9-8846-7e89d4e2a4ac.png)


This should also be the case when I want to edit my post in Gutenberg. That's why I suggested that we build logic on PHP side which tries the following:
- if it is a post type get the publish date from the post metadata
- if it is a page or custom CPT get the current month

@melchoyce does it make sense?

Reply:
Thinking a bit more about it. I'm coming to the conclusion that it might be nice to explore the following UI improvement around month and date. What if we allow the author to pick what calendar would display, some options to consider:
- inherit the post's date (this is what we have at the moment)
- use the current date
- let the author pick any date

/cc @mapk and @youknowriad for feedback